### PR TITLE
5688 breakeven new row

### PR DESCRIPTION
**Pull request by X0nic**

### CODE REVIEW

File: app/assets/javascripts/backbone/views/breakeven_product_total_view.coffee [hunk 1]
```diff
@@ -98,3 +98,12 @@ class MS.Views.BreakevenProductTotalView extends Backbone.View
     @$('.quantity').val(totals.totalQuantity.toFixed())
     @$('.revenue').val(totals.totalRevenue.toFixed())
     @$('.total_cost').val(totals.totalProductCost.toFixed())
+
+  addProduct: (product) ->
+    # Don't need to add to products since it was passed by ref
```
Comment (X0nic) [inline app/assets/javascripts/backbone/views/breakeven_product_total_view.coffee]: I could instead create a `@products` with a new array on initialize.

File: app/assets/javascripts/backbone/views/breakeven_view.coffee [hunk 2]
```diff
@@ -6,17 +6,27 @@ class MS.Views.BreakevenView extends Backbone.View
     'change .editable-table[data-table="fixed_costs"] input.amount': 'totalFixedCostsChanged'
 
   initialize: (e) ->
-    @$('.editable-breakeven-table').each (index, table) =>
-      new MS.Views.EditableBreakevenTableView(el: table)
+    tables = @$('.editable-table').map (index, table) =>
```
Comment (X0nic) [inline app/assets/javascripts/backbone/views/breakeven_view.coffee]: Refactored `EditableTable` to cover what is needed for the `EditableBreakevenTable`

File: app/assets/javascripts/backbone/views/breakeven_product_total_view.coffee [hunk 3]
```diff
@@ -98,3 +98,12 @@ class MS.Views.BreakevenProductTotalView extends Backbone.View
...
+    # @products.push(product)
+
+    product.on "product:changed", () =>
```
Comment (hooverlunch) [inline app/assets/javascripts/backbone/views/breakeven_product_total_view.coffee]: Don't need `()`. Also, please use single quotes for JS/Coffee

File: app/assets/javascripts/backbone/views/breakeven_view.coffee [hunk 4]
```diff
@@ -33,3 +43,17 @@ class MS.Views.BreakevenView extends Backbone.View
 
     @total.updated(costs)
 
+
+  addProduct: (product) ->
+    @products.push(product)
+    @total.addProduct(product)
+    product.on "BreakevenProductView:removed", (product) =>
```
Comment (hooverlunch) [inline app/assets/javascripts/backbone/views/breakeven_view.coffee]: Single quotes.

File: app/assets/javascripts/backbone/views/editable_table_view.js.coffee [hunk 5]
```diff
@@ -17,8 +17,23 @@ class MS.Views.EditableTableView extends Backbone.View
...
+      $table.append($row)
+
+    $row
 
   removeRow: (e) ->
     e.preventDefault()
     @$(e.currentTarget).closest('tr').remove()
+
+  rowAdded: ($table, $row) ->
+    @trigger('EditableTableView:rowAdded', $table, $row, @)
```
Comment (hooverlunch) [inline app/assets/javascripts/backbone/views/editable_table_view.js.coffee]: Can we use something a little more conventional for these event names, e.g. `editableTable.rowAdded`

File: app/assets/javascripts/backbone/views/breakeven_product_total_view.coffee (hunk 1 above)
Comment (hooverlunch) [inline app/assets/javascripts/backbone/views/breakeven_product_total_view.coffee]: I'm fine with this as long as it work. Please remove the commented code.

**Pull request merged**